AlphaWallet/Core/Formatters/CurrencyFormatter.swift

```diff
@@ -2,47 +2,62 @@
 
 import Foundation
 
-class CurrencyFormatter {
-    static var formatter: NumberFormatter {
-        let formatter = NumberFormatter()
-        formatter.minimumFractionDigits = 2
-        formatter.maximumFractionDigits = 2
-        formatter.roundingMode = .down
-        //TODO support multiple currency values
-        formatter.currencyCode = Currency.USD.rawValue
-        formatter.numberStyle = .currency
+extension NumberFormatter {
 
-        return formatter
+    static let currency = Formatter(.currency)
+    static let usd = Formatter(.usd)
+    static let short = Formatter(.short)
+
+    class Formatter {
+        private let formatter: NumberFormatter
+
+        fileprivate init(_ configuration: NumberFormatterConfiguration) {
+            formatter = configuration.formatter
+        }
+
+        func string(from number: NSNumber) -> String? {
+            return formatter.string(from: number)
+        }
+
+        func string(for: Any?) -> String? {
+            return formatter.string(for: number)
+        }
     }
+}
+
+private enum NumberFormatterConfiguration {
+    case short
+    case usd
+    case currency
 
-    static var usdFormatter: NumberFormatter {
+    var formatter: NumberFormatter {
         let formatter = NumberFormatter()
-        formatter.minimumFractionDigits = 2
-        formatter.maximumFractionDigits = 2
         formatter.roundingMode = .down
-        //TODO support multiple currency values
         formatter.numberStyle = .currency
-        formatter.positiveFormat = "0.00" + " " + Constants.Currency.usd
-        formatter.negativeFormat = "-0.00" + " " + Constants.Currency.usd
 
-        return formatter
-    }
+        switch self {
+        case .currency:
+            formatter.minimumFractionDigits = Constants.formatterFractionDigits
+            formatter.maximumFractionDigits = Constants.formatterFractionDigits
 
-    static var plainFormatter: EtherNumberFormatter {
-        let formatter = EtherNumberFormatter.full
-        formatter.groupingSeparator = ""
-        return formatter
-    }
+            //TODO support multiple currency values
+            formatter.currencyCode = Currency.USD.rawValue
+        case .short:
+            formatter.minimumFractionDigits = Constants.currencyFormatterFractionDigits
+            formatter.maximumFractionDigits = Constants.currencyFormatterFractionDigits
+
+            formatter.positiveFormat = "0.0000"
+            formatter.negativeFormat = "-0.0000"
+            formatter.currencyCode = String()
+        case .usd:
+            formatter.minimumFractionDigits = Constants.currencyFormatterFractionDigits
```

Comment:
This is a bit strange because `Constants.currencyFormatterFractionDigits = 4`. But it seems to work (because the result is 2 decimal places). Do you know why? And can you make it clearer?

Reply:
actually i dont know, i guess it because we set `positiveFormat/negativeFormat`, and it makes calculations depend on them.

Reply:
Very strange. Can you try to do this: https://github.com/AlphaWallet/alpha-wallet-ios/pull/1970#discussion_r448122914 and maybe the strangeness will go away. Basically keep 4 decimal places "mentioned" only in 1 formatter file and 2 decimal places in the other formatter file.